Add glossary to Trino

rosewms opened the pull request.

mosabua commented on docs/src/main/sphinx/appendix/glossary.rst:
```diff
@@ -0,0 +1,106 @@
+Glossary
+========
+
+.. _glossary:
```
we dont need this anchor since you can use `:doc:` to get here

mosabua commented on docs/src/main/sphinx/appendix/glossary.rst:
```diff
@@ -0,0 +1,106 @@
+
+The glossary contains a list of key Trino terms and definitions.
+
+.. _atoeglossary:
```
we dont need anchor for these section headers .. only for each term

rosewms commented:
> Still needs the existing terms taken out of the TLS page.

@mosabua I removed the original content and removed the extra anchors.

Ordinant commented:
> > Still needs the existing terms taken out of the TLS page.
> 
> @mosabua I removed the original content and removed the extra anchors.

Um, say what? There is all sorts of cross-referencing in tls.rst and in nearby pages TO those glossary anchors. The whole edifice was constructed carefully to enable succinct explanations without having to repeat definitions. We can't just yank that stuff out.

rosewms commented:
> > > Still needs the existing terms taken out of the TLS page.
> > 
> > 
> > @mosabua I removed the original content and removed the extra anchors.
> 
> Um, say what? There is all sorts of cross-referencing in tls.rst and in nearby pages TO those glossary anchors. The whole edifice was constructed carefully to enable succinct explanations without having to repeat definitions. We can't just yank that stuff out.

So I was able to keep the original TLS glossary term anchors. I'd added new anchors that I then removed because they were overkill for our mission.
